internal/n/crit.go

```diff
@@ -0,0 +1,94 @@
+package n
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/capatazlib/go-capataz/internal/c"
+	"github.com/capatazlib/go-capataz/internal/s"
+)
+
+// EventCriteria is an utility that allows us to specify a matching criteria to
+// a specific supervision event
+type EventCriteria func(s.Event) bool
+
+// EAnd joins a slice of EventCriteria with an and statement
+func EAnd(crits ...EventCriteria) EventCriteria {
+	return func(ev s.Event) bool {
+		result := true
+		for _, crit := range crits {
+			result = result && crit(ev)
+			if !result {
+				return result
+			}
+		}
+		return result
+	}
+}
+
+// EOr joins a slice of EventCriteria with an or statement
+func EOr(crits ...EventCriteria) EventCriteria {
+	return func(ev s.Event) bool {
+		result := false
+		for _, crit := range crits {
+			result = result || crit(ev)
+			if result {
+				return result
+			}
+		}
+		return result
+	}
+}
+
+// ENot negates the result from a given EventCriteria
+func ENot(crit EventCriteria) EventCriteria {
+	return func(ev s.Event) bool {
+		return !crit(ev)
+	}
+}
+
+// EInSubtree allows to filter s.Event that were sent from an specific subtree
```

Comment:
hmm ... just thinking out-loud, the list makes it look like you are sending a list of subtrees. I like the apparent convenience of not having to build the long name, but it may obscure the real _name_ of the path. The name includes the `s.NodeSepToken`. Part of me thinks that you should have a convenience function to build the path from the constituents and here you should take the actual name 🤷‍♂️ . Just a thought.